helpers/functions.py

```diff
@@ -306,3 +306,41 @@ def levered_beta(unlevered_beta:float,tax_rate:float,debt:float,equity:float):
     l_beta = unlevered_beta*(1+(1-tax_rate)*(debt//equity))
     return l_beta
 
+#Function to calculate Duration With Convexity Adjustment
+def duration(rate:float,coupon_rate:float,frequency:float,face_value:float,settlement_date:float,maturity_date:float):
+    import numpy as np 
```

Comment:
As you are using external libraries, make sure to include it in the `requirements.txt` file. So that other users can install this dependency too.